### PR TITLE
Rewrite README

### PR DESCRIPTION
**Pull request by rouille**
[Pull Request doc](https://breakthrough-energy.github.io/docs/user/git_guide.html#d-pull-request)

### Purpose
Move sections of README in ReST files located in a new **docs** folder and rewrite README.

### What the code is doing
There is no code

### Testing
I have checked that the content of the **powersimdata/docs** folder is properly display in our docs. To reproduce, checkout this branch and the `ben/powersimdata` branch in the ***docs*** repo and run `tox` there to make sure that the PowerSimData tutorial displays correctly

### Where to look
The content of the ReST files is almost identical to the sections in the current README. The only changes I made are:
* change code snippet to use #420 
* rearrange subsections in the scenario framework section to have create --> execute --> analyze --> delete --> move

The new README is the file that requires attention  

### Time estimate
Hard to say

### CODE REVIEW

File: docs/scenario.rst [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+    # print name of Scenario object state
+    print(scenario.state.name)
+
+    # Start building a scenario
+    scenario.set_grid(grid_model="usa_tamu", interconnect="Western")
+
+    # set plan and scenario names
+    scenario.set_name("test", "dummy")
+    # set start date, end date and interval
+    scenario..set_time("2016-08-01 00:00:00", "2016-08-31 23:00:00", "24H")
```
Comment (danielolsen) [inline docs/scenario.rst:76]: `scenario..set_time` -> `scenario.set_time`

File: README.md [hunk 2]
```diff
@@ -1,470 +1,73 @@
...
-
-The list of targets may be specified in either a CSV file or a data frame, as long as the required columns are present: `region_name` and `ce_target_fraction`. Optional columns are: `allowed_resources` (defaulting to solar & wind), `external_ce_addl_historical_amount` (clean energy not modeled in our grid, defaulting to 0), and `solar_percentage` (how much of the new capacity will be solar, defaulting to the current solar:wind ratio. This input only applies to the *independent* strategy, a shared-goal new solar fraction for *collaborative* planning is specified in the function call to `calculate_clean_capacity_scaling`.
+## Dependencies
+**PowerSimData** relies on several Python all available on [PyPi](https://pypi.org/).
```
Comment (kasparm) [inline README.md:29]: Python packages you mean?

File: docs/scenario.rst (hunk 1 above)
Comment (rouille) [reply docs/scenario.rst]: Done

File: README.md (hunk 2 above)
Comment (rouille) [reply README.md]: Done

**Comment by rouille** [comment]
Do not hesitate to propose improvements for the README, e.g., more or different bullet points in the **Main Features**, etc. section)

**Comment by kasparm** [comment]
Looks good @rouille. I like the the tutorial. I created the docs and it displays correctly. 
Out of scope: Diagram for software packages and how they interact.

**Review by kasparm — APPROVED** [review]

**Comment by rouille** [comment]
> Looks good @rouille. I like the the tutorial. I created the docs and it displays correctly.
> Out of scope: Diagram for software packages and how they interact.

We might want to use [mermaid](https://pypi.org/project/sphinxcontrib-mermaid/) for that

**Comment by BainanXia** [comment]
> > Looks good @rouille. I like the the tutorial. I created the docs and it displays correctly.
> > Out of scope: Diagram for software packages and how they interact.
> 
> We might want to use [mermaid](https://pypi.org/project/sphinxcontrib-mermaid/) for that

Diagrams can be generated automatically? This is fancy!

**Comment by danielolsen** [comment]
Is ReST better than markdown? Should we write new READMEs in ReST?

**Comment by rouille** [comment]
> Is ReST better than markdown? Should we write new READMEs in ReST?

ReST is the default plaintext markup language used by Sphinx. So far we have used a plugin to display Markdown file in order linked to our package's README. Since, we want i) to have concise README and ii) take full advantage of the more elaborate ReST language, I have translated all the Markdown files to ReST

File: README.md [hunk 3]
```diff
@@ -1,470 +1,74 @@
...
-This package has been written in order to carry out power flow study in the U.S. electrical grid. This framework allows the user to easily build extensive scenarios.
+**PowerSimData** is part of a Python software ecosystem developed by [Breakthrough
+Energy Sciences](https://science.breakthroughenergy.org/) to carry out power flow study
+in the U.S. electrical grid.
 
-PowerSimData is part of a set of packages representing Breakthrough Energy's power system model. More information regarding the installation of the model as well as the contribution guide can be found [here](https://breakthrough-energy.github.io/docs/).
 
+## Main Features
+Here are a few things that **PowerSimData** can do:
+* Provide a flexible modeling to create complex scenarios
```
Comment (jon-hagg) [inline README.md:15]: Should it be "flexible modeling tool" or "flexible model"?

Comment (rouille) [reply README.md:15]: Thanks. I like flexible modeling tool

File: README.md [hunk 4]
```diff
@@ -1,470 +1,74 @@
...
+* Perform investment cost studies
+* Call simulation engine of power system operation to run power flow study
```
Comment (jon-hagg) [inline README.md:17]: Might be clearer to say something like "Run power flow study using interface to external simulation engine"

File: README.md [hunk 5]
```diff
@@ -1,470 +1,74 @@
...
+* Retrieve simulation outputs
+* Handle scenario data efficiently
```
Comment (jon-hagg) [inline README.md:19]: I might combine the last 2 bullet points to "Manage data throughout the lifecycle of a simulation"

File: README.md [hunk 6]
```diff
@@ -1,470 +1,74 @@
...
-
-The list of targets may be specified in either a CSV file or a data frame, as long as the required columns are present: `region_name` and `ce_target_fraction`. Optional columns are: `allowed_resources` (defaulting to solar & wind), `external_ce_addl_historical_amount` (clean energy not modeled in our grid, defaulting to 0), and `solar_percentage` (how much of the new capacity will be solar, defaulting to the current solar:wind ratio. This input only applies to the *independent* strategy, a shared-goal new solar fraction for *collaborative* planning is specified in the function call to `calculate_clean_capacity_scaling`.
+## Dependencies
+**PowerSimData** relies on several Python packages all available on
+[PyPi](https://pypi.org/). The list can be found in the ***requirements.txt*** or
+****Pipfile*** files both located at the root of this package.
```
Comment (jon-hagg) [inline README.md:31]: Extra `*` on left of Pipfile

File: README.md (hunk 5 above)
Comment (rouille) [reply README.md:19]: Nice

File: README.md [hunk 7]
```diff
@@ -1,470 +1,74 @@
...
-
-
-### B. Analysis of Generation Upgrades
-#### I. Accessing and Saving Relevant Supply Information
-Analyzing generator supply and cost curves requires the proper generator cost and plant information to be accessed from a Grid object. This data can be accessed using the following:
-```python
-from powersimdata.design.generation.cost_curves import get_supply_data
-
-supply_df = get_supply_data(grid, num_segments, save)
+Only a limited set of feature is available when solely installing **PowerSimData**. If you chose this option, we recommend that you use `pipenv`:
```
Comment (jon-hagg) [inline README.md:42]: "..set of features are available..", and "If you choose this option"

File: README.md (hunk 3 above)
Comment (rouille) [reply README.md]: Done

File: README.md (hunk 5 above)
Comment (rouille) [reply README.md]: done

File: README.md (hunk 7 above)
Comment (rouille) [reply README.md]: Done

**Review by jon-hagg — APPROVED** [review]

File: docs/scenario.rst [hunk 8]
```diff
@@ -0,0 +1,316 @@
...
+.. code-block:: python
+
+    process_run = scenario.launch_simulation(extract_data=False)
+    # Extract data
+    process_extract = scenario.extract_simulation_output()
+
+
+Retrieving Scenario Output Data
++++++++++++++++++++++++++++++++
+When the ``Scenario`` object is in the **analyze** state, the user can access various
```
[the remark this answers is not in the document]
Comment (rouille) [reply docs/scenario.rst:212]: Yes

[the remark this answers is not in the document]
Comment (rouille) [reply docs/scenario.rst:212]: I plan to include the following sentence right above the `Retrieving Scenario Output Data`:
```
Note that you will need to create a new ``Scenario`` object via the scenario id/name to
access the output data.
```
What do you think?

[the remark this answers is not in the document]
Comment (ahurli) [reply docs/scenario.rst:212]: I like it!

File: docs/grid.rst [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+a combination of two interconnections.
+
+A ``Grid`` object can be created as follows:
+
+- U.S. grid
+
+  .. code-block:: python
+
+    from powersimdata.input.grid import Grid
+    usa = Grid("Western")
```
[the remark this answers is not in the document]
Comment (rouille) [reply docs/grid.rst]: Done

File: docs/scenario.rst (hunk 8 above)
[the remark this answers is not in the document]
Comment (rouille) [reply docs/scenario.rst:217]: Done

**Pull request merged**